client-hc/src/main/java/com/graphhopper/api/MatrixResponse.java

```diff
@@ -128,7 +128,7 @@ public double getDistance(int from, int to) {
         } else if (to >= distances[from].length) {
             throw new IllegalStateException("Cannot get 'to' " + to + " from distances with size " + distances[from].length);
         }
-        return distances[from][to];
+        return distances[from][to] == Integer.MAX_VALUE ? Double.MAX_VALUE : distances[from][to];
```

Comment:
oh, thx - this is a good catch :+1:

Reply:
Shouldn't it better map to `infinity` instead of Double.MAX_VALUE?

Reply:
Or the other way around: disconnected == i.e. "infinity" gets serialized into `null` and should be back in Java world infinity (?)

Reply:
Yes previously this case never occurred because there would have been an error if points were not connected. Now we have this case if failFast==false, yes. Hm not sure about infinity vs. max. For time we also return `Long.MAX_VALUE`.

Reply:
The 'impossible' weight is `Double.MAX_VALUE` on the server side as well (so far). The same for `Path#weight`:

https://github.com/graphhopper/graphhopper/blob/c43d03e9cbc2690b1c3bef319921f6f15adc5d82/core/src/main/java/com/graphhopper/routing/Path.java#L72-L74

Reply:
Ah, yes. For `long` there is not infinity. Ugly, as we currently use "infinity" for "disconnected" in the Weighting and so I thought we could make it more consistent (and maybe refactor `Path#weight` too) but this would make it inconsistent with `long`.

Reply:
Do you prefer one over the other, I do not have a strong opinion about this and either way is fine for me.

Reply:
> because normally when hasProblems()==true we can use getDisconnectedPoints() to check which connections are missing (instead of checking getWeight(3, 5) == Double.MAX_VALUE`

This seems like the best solution to force or promote the usage of disconnected points in this cases (or have a convenience method like `isConnected(from, to)`)

Reply:
Yes having `isConnected(from, to)` is a very good idea, if we have this there should not really be a reason to check something like `getWeight(3,5)==Double.MAX_VALUE` and we only do this internally making the concrete 'impossible value' even 'less important'. Will do.

Reply:
Nice, thanks!